backend/src/api/endpoints/document_tag.py

```diff
@@ -23,28 +26,50 @@
 @router.put(
     "",
     response_model=DocumentTagRead,
-    summary="Creates a new DocumentTag",
-    description="Creates a new DocumentTag and returns it with the generated ID.",
+    summary="Creates a new DocumentTag and returns it with the generated ID.",
 )
-async def create_new_doc_tag(
-    *, db: Session = Depends(get_db_session), doc_tag: DocumentTagCreate
+def create_new_doc_tag(
+    *,
+    db: Session = Depends(get_db_session),
+    doc_tag: DocumentTagCreate,
+    authz_user: AuthzUser = Depends(),
+    validate: Validate = Depends(),
 ) -> DocumentTagRead:
+    authz_user.assert_in_project(doc_tag.project_id)
+
+    if doc_tag.parent_tag_id is not None and doc_tag.parent_tag_id != -1:
+        authz_user.assert_in_same_project_as(Crud.DOCUMENT_TAG, doc_tag.parent_tag_id)
+
+        parent_tag = crud_document_tag.read(db, doc_tag.parent_tag_id)
+        validate.validate_condition(
+            parent_tag.project_id == doc_tag.project_id,
+            "Parent tag needs to be in the same project",
+        )
+
     db_obj = crud_document_tag.create(db=db, create_dto=doc_tag)
     return DocumentTagRead.model_validate(db_obj)
 
 
 @router.patch(
     "/bulk/link",
     response_model=int,
-    summary="Links multiple DocumentTags with the SourceDocuments",
-    description="Links multiple DocumentTags with the SourceDocuments and returns the number of new Links",
+    summary="Links multiple DocumentTags with the SourceDocuments and returns the number of new Links",
 )
-async def link_multiple_tags(
+def link_multiple_tags(
     *,
     db: Session = Depends(get_db_session),
     multi_link: SourceDocumentDocumentTagMultiLink,
+    authz_user: AuthzUser = Depends(),
 ) -> int:
-    # TODO Flo: only if the user has access?
+    # TODO this is a little inefficient, but at the moment
+    # the fronend is never sending more than one id at a time
+    authz_user.assert_in_same_project_as_many(
+        Crud.SOURCE_DOCUMENT, multi_link.source_document_ids
+    )
+    authz_user.assert_in_same_project_as_many(
+        Crud.DOCUMENT_TAG, multi_link.document_tag_ids
+    )
+
```

Comment:
The sdoc and tags should be in the same project. That is, it should not be possible to add a tag from project A to a sdoc from project B

Reply:
Since a user can be in projects A and B, we have to assert that explicitly.

Reply:
Fixed